src/nom.rs

```diff
@@ -0,0 +1,201 @@
+use fxhash::FxHashSet;
+use serde::ser::{SerializeStruct, Serializer};
+use serde::Serialize;
+use std::fmt;
+use tree_sitter::Node;
+
+use crate::checker::Checker;
+
+use crate::*;
+
+#[derive(Default, Debug)]
+pub struct Stats {
+    functions: FxHashSet<String>,
+    closures: usize,
+}
+
+impl Serialize for Stats {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut st = serializer.serialize_struct("nom", 3)?;
+        st.serialize_field("functions", &self.functions())?;
+        st.serialize_field("closures", &self.closures())?;
+        st.serialize_field("total", &self.total())?;
+        st.end()
+    }
+}
+
+impl fmt::Display for Stats {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "functions: {}, \
+             closures: {}, \
+             total: {}",
+            self.functions(),
+            self.closures(),
+            self.total(),
+        )
+    }
+}
+
+impl Stats {
+    pub fn merge(&mut self, other: &Stats) {
+        for func_identifier in other.functions.iter() {
+            self.functions.insert(func_identifier.to_string());
+        }
+        self.closures += other.closures;
+    }
+
+    #[inline(always)]
+    pub fn functions(&self) -> f64 {
+        self.functions.len() as f64
+    }
+
+    #[inline(always)]
+    pub fn closures(&self) -> f64 {
+        self.closures as f64
+    }
+
+    #[inline(always)]
+    pub fn total(&self) -> f64 {
+        self.functions() + self.closures()
+    }
+}
+
+pub trait Nom
+where
+    Self: Checker,
+{
+    fn compute(_node: &Node, _code: &[u8], _stats: &mut Stats) {}
+}
+
+impl Nom for PythonCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Python::*;
+
+        match node.kind_id().into() {
+            FunctionDefinition => {
+                stats.functions.insert(std::string::String::new());
```

Comment:
Why do you insert an empty string here ?
The count wil be 0 or 1.

Reply:
I've used `String` in the `FxHashSet` instead of numbers because I wanted to avoid this case:

```
trait Foo {
 fn foo();
}

impl Foo for Test {
  fn foo() {
    let a = 0;
  }
}
```

In this case there would be 2 `foo` identifier for the unit space, and that would be wrong because the two `foo` are the same. One is the declaration and one its implementation. 

Since we have this case for `Rust` and not for other languages, I inserted an empty string when a method is encountered.

Reply:
If you run rca -m on:
```python
def a():
    pass

def b():
    pass

def c():
    pass
```
you'll get nom::functions=1 which is wrong, correct ?

Reply:
And if I dump the ast for your rust example (`rca -d`), we can see that the parent node for `fn foo();` is a `function_signature_item` and for the other `foo` it's a `function_item`. So you should be able to discriminate definition vs declaration using the context.

Reply:
Yep, you're right. It's wrong. I will update that. Thanks!